site/examples/check-lists.js

```diff
@@ -32,7 +32,39 @@ const CheckListsExample = () => {
 }
 
 const withChecklists = editor => {
-  const { deleteBackward } = editor
+  const { deleteBackward, insertBreak } = editor
+  editor.insertBreak = () => {
+    const match = Editor.above(editor, {
+      match: n => {
+        return Editor.isBlock(editor, n)
+      },
+    })
+    if (match) {
+      const [matchingNode] = match
+      if (matchingNode.type !== 'paragraph') {
+        if (
+          matchingNode.children[matchingNode.children.length - 1].text
+            .length === 0
+        ) {
+          Transforms.setNodes(editor, {
+            type: 'paragraph',
+            children: [{ text: '' }],
+          })
+          return
+        }
+        if (matchingNode.type === 'check-list-item') {
+          const checklist = {
+            type: 'check-list-item',
+            checked: false,
+            children: [{ text: '' }],
```

Comment:
This logic doesnt handle the case where the checklist item block is being split in when selection is collapsed in the middle. The end half of the line is truncated instead of the text being carried down to the new block.

Reply:
@CameronAckermanSEL  I agree the text doesn't get carried forward, its a bug, thanks for finding that out. Any idea on how can I handle that case?

Reply:
@arpit016 have you tried overriding splitNodes instead?

Reply:
@CameronAckermanSEL  splitNodes will work if the selection or the cursor is in between the text, but if its at the end of the text, it wont work, so is there a method where I can check that the enter pressed is from the end of the text and not from between the text?

Reply:
@CameronAckermanSEL  I think I figured it out, updating the pull request in few minutes

Reply:
@CameronAckermanSEL I have updated the code to handle the bug that was there. Please checka nd review once